.github/workflows/release.yaml

```diff
@@ -30,17 +30,17 @@ jobs:
     needs:
       - lib-check
       - ci-tests
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
```

Comment:
I would like to run 22.04 (which is ubuntu-latest at the moment), but GH is not happy about it (see the commit message):
https://github.com/canonical/mysql-router-k8s-operator/pull/9/commits/1e72239f1e205a4b6bdee571a09a138ac4121873

The link to failing action: https://github.com/canonical/mysql-router-k8s-operator/actions/runs/3830362423/jobs/6518485060
Good ideas are welcome!

Reply:
I forget to mention, all other CI configs are fine with 22.04 (ubuntu-latest), but charming-actions/upload-charm.

After the second look I noticed '--destructive-mode' in https://github.com/canonical/mysql-router-k8s-operator/actions/runs/3830362423/jobs/6518485060

> --destructive-mode:  Pack charm using current host ...

So, the host MUST match build-on in charmcraft.yaml. Resolved?

Reply:
Build-on=22.04 and run-on=20.04 might be an option (with some new and unpredictable results).
At the moment we need to update 20.04 version to test Mykola's refactoring without new issues of cross-building.

Shayan recommended to use mysql-bundle approach (call charmcraft manually without '--destructive-mode'): https://github.com/canonical/mysql-bundle/blob/main/Makefile#L21

The good question is: why '--destructive-mode' is mandatory for 'charming-actions'? Can we make it optional?
It will increase usability for us as we want to provide several run-on versions for subordinate charms.

Reply:
I believe it's used to speed things up. Destructive mode means "don't use a LXD container, just run the build on this host" - which is fine in almost all cases in a CI environment where you don't care too much for the long term integrity of the host (because there is no long-term ;-))

@simskij may be able to give more background, and perhaps we do need to consider allowing this to be optional.

For now, while it's a workaround, I don't see a reason why we need to test/maintain `runs-on==20.04`, we should really be targeting 22.04 in the `charmcraft.yaml`:

```
type: charm
bases:
  - build-on:
      - name: "ubuntu"
        channel: "22.04"
    run-on:
      - name: "ubuntu"
        channel: "22.04"
parts:
  charm:
    charm-binary-python-packages:
      - mysql-connector-python==8.0.30
```

Reply:
Updated build-on/run-on to 22.04.

Reply:
Reported `upload-charm` action improvement as https://github.com/canonical/charming-actions/issues/80